App.tsx

```diff
@@ -76,21 +107,54 @@ class App extends Component<{}, appState> {
     });
   };
 
+  success = (location: any) => {
+    const { position } = this.state;
+    position.setLatitude(location.coords.latitude);
+    position.setLongitude(location.coords.longitude);
+    this.setMapLocation(position);
+  };
+
+  failure = (err: any) => {
+    console.log(err);
+  };
+
+  displayCurrentLocation() {
+    const options = {
+      enableHighAccuracy: true,
+      timeOut: 20000,
+      maximumAge: 60 * 60 * 24,
+    };
+    navigator.geolocation.getCurrentPosition(this.success, this.failure, options);
+  }
+
   render() {
     const { region, markers, polygons } = this.state;
 
     return (
       <View style={styles.container}>
         <View style={styles.search} />
         <CampusToggleButton setMapLocation={this.setMapLocation} />
-        <MapView provider={PROVIDER_GOOGLE} style={styles.mapStyle} region={region}>
+        <MapView
+          provider={PROVIDER_GOOGLE}
+          style={styles.mapStyle}
+          region={region}
+          showsUserLocation
+          showsMyLocationButton={false}
+          onUserLocationChange={() => this.updateLocation}
+          showsCompass={false}
+        >
           <PolygonsAndMarkers markers={markers} polygons={polygons} />
           <ShowDirection
             startLocation={new OutdoorPOI(new Location(45.458488, -73.639862), 'test-start')}
             endLocation={new OutdoorPOI(new Location(45.50349, -73.572182), 'test-end')}
             transportType={transportMode.transit}
           />
         </MapView>
+        <View style={styles.positionBtn}>
+          <TouchableOpacity onPress={() => this.displayCurrentLocation()}>
+            <Image style={styles.iconSize} source={require('./assets/cp.png')} />
+          </TouchableOpacity>
+        </View>
```

Comment:
This should be a separate component called 'GetLocationButton' and you would pass the 'setMapLocation' function from App.tsx. Look at my ToggleCampusButton Component for reference. Yours should look something like 

<GetLocationButton setMapLocation={this.setMapLocation} />

The purpose of this is to really make our app into components which helps with building logic